core/server/models/plugins/order.js

```diff
@@ -31,11 +31,16 @@ const order = function order(Bookshelf) {
                 field = match[1].toLowerCase();
                 direction = match[2].toUpperCase();
 
-                if (orderAttributes.indexOf(field) === -1) {
+                const fieldRegexp = new RegExp(`${field}$`);
```

Comment:
Yupp, the format of ordeing fields will be `{table_name}.{orderable_field_name}`. The regexp tries to match with the ending of the name. An alternative would be doing a match on the `field === orderAttribute.split('.')[1]`. This alternative method might be less performant I think, but there's rarely more than couple dozen records expected in these "orderAttributes", so maybe worth going with readability instead of "magic-regexp" :thinking: Wdyt? 

Side-note, both methods have a limitation that it not obvious and probably needs a `//NOTE:`. It's about not being able to use fields with the same name in "native" and "related" tables (for example if both `posts` and `posts_meta` tables had `title` field, it would be impossible to order by `title`, but that's expected I think given ordering is done of resource's fields which are usually unique). There is no usecase for this yet, but might avoid some confusion later on.

Reply:
Could you use  ```value.endsWith(`.${field}`)``` in the find fn instead of splitting the string?

Reply:
Good point @vikaspotluri123 ! Why is my brain coming up with complex solutions first when there's something so much easier :sweat_smile:

Reply:
>  It's about not being able to use fields with the same name in "native" and "related" tables

This is what I was thinking, are you able to access the table name from the plugin, to get around this? I'm wondering if we would have issues with the order of the `orderAttributes` array?

e.g. `['parent.title', 'child.title']` and `['child.title', 'parent.title']` would give different orderings

Can you guarantee the order of that array? Should we consider `sort`ing it so that we can?

Reply:
> e.g. ['parent.title', 'child.title'] and ['child.title', 'parent.title'] would give different orderings

Yupp, exactly the problem we _might_ face in the future. My thinking at the moment is that incoming order parameters will always be unique and "YAGNI" rule. I'd leave an extended comment about the limitations for now and move on.

> Can you guarantee the order of that array?

"Fun" fact - we can't with current implementation :grimacing: (raising an issue for this because it will probably bite us) What I have discovered is we keep a hash instead of an array in `options.order` attribute and than [iterate over hash keys](https://github.com/TryGhost/Ghost/blob/2cbba946632cb658d179861b4bb9fff697407441/core/server/models/plugins/pagination.js#L181-L186) when ordering is added to the query.  
For future reference, illustration of why ordering is not guaranteed with object keys even if defined in certain order ([source](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)):
```
// array-like object with random key ordering
const anObj = { 100: 'a', 2: 'b', 7: 'c' };
console.log(Object.keys(anObj)); // console: ['2', '7', '100']
```

> Should we consider sorting it so that we can?

We :100:  should once the above problem is solved and we have a clear usecase in our resources/models.

Reply:
I've tried creating some test cases to confirm the case or ignored order query parameters order, but seems like because we use lodash's `_.forOwn` the order has been kept correctly. That said still think that keeping information about something that has to maintain it's order in a hash is not a best choice, arrays should be used if the order of properties is something we care about.